src/syscall_hook.h

```diff
@@ -0,0 +1,28 @@
+#ifndef CORE_SYSCALL_HOOK_H
+#define CORE_SYSCALL_HOOK_H
+#include <sys/socket.h>
+
+typedef int (*sys_connect)(int sockfd,
+                           const struct sockaddr *addr,
+                           socklen_t addrlen);
+typedef int (*sys_accept)(int sockfd,
+                          struct sockaddr *addr,
+                          socklen_t *addrlen);
+
+typedef ssize_t (*sys_read)(int fd, void *buf, size_t count);
+typedef ssize_t (*sys_recv)(int sockfd, void *buf, size_t len, int flags);
+
+typedef ssize_t (*sys_write)(int fd, const void *buf, size_t count);
+typedef ssize_t (*sys_send)(int sockfd, const void *buf, size_t len, int flags);
+
+/* declared in syscall_hook.c */
+extern sys_connect real_sys_connect;
```

Comment:
At present, there should be only an extern variable with prefix `real_sys_`. i.e., `real_sys_write`.
Others should be kept locally in file `src/syscall_hook.c` for the sake of further compiler optimizations.